pandas/tests/io/excel/test_readers.py

```diff
@@ -712,7 +712,9 @@ def test_dtype_backend_string(self, read_ext, string_storage, tmp_excel):
                 "b": Series(["x", None], dtype=pd.StringDtype(string_storage)),
             }
         )
-        tm.assert_frame_equal(result, expected)
+        # the storage of the str columns' Index is also affected by the
+        # string_storage setting -> ignore that for checking the result
+        tm.assert_frame_equal(result, expected, check_column_type=False)
```

Comment:
This changes the tests to essentially just ignore the issue.

We could also verify it explicitly (although it's not the main thing that this test is meant to test) by doing something like:

```python
if using_infer_string:
    expected.columns = expected.columns.astype(pd.StringDtype(string_storage, na_value=np.nan))
```

Which, now I comment that here, is maybe also perfectly fine (not too verbose, compared to the comment I had to add :))